toolchains/vivado.py

```diff
@@ -329,3 +331,27 @@ def add_runtimes(self):
 
     def versions(self):
         return {'yosys': self.yosys_ver(), 'vivado': self.vivado_ver()}
+
+
+class VivadoYosysUhdm(VivadoYosys):
+    '''Vivado PnR + Yosys synthesis using uhdm frontend'''
+    def __init__(self, rootdir):
+        VivadoYosys.__init__(self, rootdir)
+        self.synthtool = 'yosys'
+        self.synthoptions = [
+            '-iopad', '-arch xc7', '-flatten', 'frontend=surelog'
+        ]
+        self.toolchain = 'yosys-vivado-uhdm'
+        # by default assume, uhdm-yosys is installed in symbiflow env
+        uhdm_yosys_path = f"{self.rootdir}/env/conda/envs/symbiflow-env/share"
+        for path in os.environ["PATH"].split(os.pathsep):
+            exe_file = os.path.join(path, "uhdm-yosys")
+            if os.path.isfile(exe_file) and os.access(exe_file, os.X_OK):
+                uhdm_yosys_path = path
+                break
```

Comment:
we may use `which uhdm-yosys` to get the location of the binary here, without assuming it from conda or the PATH